### PR TITLE
Kutter rest api search objects

### PR DESCRIPTION
**Pull request by Chufle**
change-log:
- new lambda function search_objects(), search over metadata "originalFileName" and "creationDate", will be extended soon
- API endpoint for search_objects():
`https://***.execute-api.us-west-2.amazonaws.com/prod/search-objects?search=***`

### CODE REVIEW

File: search_objects/src/search_objects.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    filter_expression = Or( Attr('creationDate').contains(search), Attr('originalFileName').contains(search))
+    response = table.scan(Limit = limit,FilterExpression = filter_expression) 
+    return { 
+         'items' :response['Items']
+    }
+
+def handler(event, context):
+    search = event ['queryStringParameters']['search']
+    limit = int(10)
+    items = search_objects_data(search,limit)
```
Comment (fabianschmauder) [inline search_objects/src/search_objects.py:18]: you can also add pagination instead of fixt limit

File: search_objects/src/search_objects.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import json
+import boto3
+from boto3.dynamodb.conditions import Attr, Or
+
+dynamodb = boto3.resource('dynamodb')
+table = dynamodb.Table('kutter-table')
+
+def search_objects_data(search, limit):
+    filter_expression = Or( Attr('creationDate').contains(search), Attr('originalFileName').contains(search))
+    response = table.scan(Limit = limit,FilterExpression = filter_expression)
```
Comment (fabianschmauder) [inline search_objects/src/search_objects.py:10]: when you filter for creation date i would expect that you can define a range

**Review by fabianschmauder — COMMENTED** [review]
nice , just pagination and filter for date range

File: search_objects/src/search_objects.py (hunk 1 above)
Comment (Chufle) [reply search_objects/src/search_objects.py:18]: will be tracked in issue https://github.com/Chufle/kutter/issues/68

File: search_objects/src/search_objects.py (hunk 2 above)
Comment (Chufle) [reply search_objects/src/search_objects.py:10]: will be tracked in issue https://github.com/Chufle/kutter/issues/69

**Pull request merged**